### PR TITLE
Add IXCLRDataProcess::GetRuntimeNameByAddress Method

### PR DESCRIPTION
**Pull request by tommcdon**
## Summary

Add IXCLRDataProcess::GetRuntimeNameByAddress Method

### CODE REVIEW

File: docs/framework/unmanaged-api/debugging/ixclrdataprocess-getruntimenamebyaddress-method.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    [out] ULONG32 *nameLen,
+    [out, size_is(bufLen)] WCHAR nameBuf[],
+    [out] CLRDATA_ADDRESS* displacement
+);
+```
+
+## Parameters
+
+`address`\
+[in] A CLRDATA_ADDRESS that stores the virtual memory address.
```
Comment (sdmaclea) [inline docs/framework/unmanaged-api/debugging/ixclrdataprocess-getruntimenamebyaddress-method.md:41]: This is unclear.  What is the address of?

File: docs/framework/unmanaged-api/debugging/ixclrdataprocess-getruntimenamebyaddress-method.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+[in] The length of the buffer.
+
+`namLen`\
+[out] A pointer to the number of characters returned.
+
+`namBuf`\
+[out, size_is(`bufLen`)] The input buffer of length `bufLen`.
+
+`displacement`\
+[out] A pointer to the code offset.
```
Comment (sdmaclea) [inline docs/framework/unmanaged-api/debugging/ixclrdataprocess-getruntimenamebyaddress-method.md:56]: This could use a bit more explanation.  What code?

**Comment by sdmaclea** [comment]
Markdown lint is failing due to trailing spaces
```
docs/framework/unmanaged-api/debugging/ixclrdataprocess-getruntimenamebyaddress-method.md:44:16 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
docs/framework/unmanaged-api/debugging/ixclrdataprocess-getruntimenamebyaddress-method.md:60:126 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
```

File: docs/framework/unmanaged-api/debugging/ixclrdataprocess-getruntimenamebyaddress-method.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
+---
+title: "IXCLRDataProcess::GetRuntimeNameByAddress Method"
```
Comment (gewarren) [inline docs/framework/unmanaged-api/debugging/ixclrdataprocess-getruntimenamebyaddress-method.md:2]: This new file should be linked from toc.yml too.

File: docs/framework/unmanaged-api/debugging/ixclrdataprocess-getruntimenamebyaddress-method.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+);
+```
+
+## Parameters
+
+`address`\
+[in] A CLRDATA_ADDRESS that stores the virtual memory address.
+
+`flags`\
+[in] Set to '0'
```
Comment (gewarren) [inline docs/framework/unmanaged-api/debugging/ixclrdataprocess-getruntimenamebyaddress-method.md:44]:
```suggestion
[in] Set to '0'.
```

File: docs/framework/unmanaged-api/debugging/ixclrdataprocess-getruntimenamebyaddress-method.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+`namBuf`\
+[out, size_is(`bufLen`)] The input buffer of length `bufLen`.
+
+`displacement`\
+[out] A pointer to the code offset.
+
+## Remarks
+
+The provided method is part of the `IXCLRDataProcess` interface and corresponds to the 16th slot of the virtual method table.
```
Comment (gewarren) [inline docs/framework/unmanaged-api/debugging/ixclrdataprocess-getruntimenamebyaddress-method.md:60]:
```suggestion
The provided method is part of the `IXCLRDataProcess` interface and corresponds to the 16th slot of the virtual-method table.
```

File: docs/framework/unmanaged-api/debugging/ixclrdataprocess-getruntimenamebyaddress-method.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+> [!NOTE]
+Returns S_FALSE if the buffer is not large enough for the name, and sets nameLen to be the buffer length needed.
```
Comment (gewarren) [inline docs/framework/unmanaged-api/debugging/ixclrdataprocess-getruntimenamebyaddress-method.md:63]:
```suggestion
> If the buffer is not large enough for the name, this method returns `S_FALSE` and sets `nameLen` to the required buffer length.
```

File: docs/framework/unmanaged-api/debugging/ixclrdataprocess-getruntimenamebyaddress-method.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+
+## Requirements
+
+**Platforms:** See [System Requirements](../../../../docs/framework/get-started/system-requirements.md).
```
Comment (gewarren) [inline docs/framework/unmanaged-api/debugging/ixclrdataprocess-getruntimenamebyaddress-method.md:67]:
```suggestion
**Platforms:** See [System Requirements](../../get-started/system-requirements.md)
```

**Review by gewarren — COMMENTED** [review]
Thanks for adding this.

File: docs/framework/unmanaged-api/debugging/ixclrdataprocess-getruntimenamebyaddress-method.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+**Header:** None
```
Comment (gewarren) [inline docs/framework/unmanaged-api/debugging/ixclrdataprocess-getruntimenamebyaddress-method.md:68]:
```suggestion
**Header:** None\
```

File: docs/framework/unmanaged-api/debugging/ixclrdataprocess-getruntimenamebyaddress-method.md [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+**Library:** None
```
Comment (gewarren) [inline docs/framework/unmanaged-api/debugging/ixclrdataprocess-getruntimenamebyaddress-method.md:69]:
```suggestion
**Library:** None\
```

**Review by gewarren — COMMENTED** [review]
Add a couple more suggestions to add hard line breaks.

**Comment by IEvangelist** [comment]
Hi @tommcdon - just working backwards through open PRs. Any updates on this PR?

**Comment by tommcdon** [comment]
> Hi @tommcdon - just working backwards through open PRs. Any updates on this PR?

My apologies.  I just pushed changes for the PR feedback.

**Review by gewarren — APPROVED** [review]

File: docs/framework/unmanaged-api/debugging/ixclrdataprocess-getruntimenamebyaddress-method.md [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+
+`displacement`\
+[out] A `CLRDATA_ADDRESS` pointer to the code offset of the returned symbol.
+
+## Remarks
+
+The provided method is part of the `IXCLRDataProcess` interface and corresponds to the 16th slot of the virtual-method table.
+
+> [!NOTE]
+If the buffer is not large enough for the name, this method returns `S_FALSE` and sets `nameLen` to the required buffer length.
```
Comment (gewarren) [inline docs/framework/unmanaged-api/debugging/ixclrdataprocess-getruntimenamebyaddress-method.md:63]:
```suggestion
> If the buffer is not large enough for the name, this method returns `S_FALSE` and sets `nameLen` to the required buffer length.
```

**Pull request merged**